### PR TITLE
test(flat table): add refactored tests for flat table FE-4540

### PR DESCRIPTION
**Pull request by sianford**
### Proposed behaviour

Refactor Cypress tests for Flat Table component to use new cypress-component-react framework for testing. 
Add flat-table.test.js file. Delete flat-table.feature file and all associated step definitions and json files.

### Current behaviour

Current process for Flat Table is the Cypress approach with BDD.

### Checklist

- [ ] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run npx cypress open --component to check there is a newly added test.file
- [x] Run the flat-table.test.js file and check it passes
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Check no other BDD style files remain
- [x] Run npx cypress run --component to check none of the other *.test.js files have regressed
- [x] Run npx cypress run --e2e to check none of the remaining accessibility files have regressed

### CODE REVIEW

File: src/components/flat-table/flat-table.test.js [hunk 1]
```diff
@@ -0,0 +1,4635 @@
...
+    );
+
+    it("should render Flat Table with head and body nodes as children", () => {
+      CypressMountWithProviders(<FlatTableComponent />);
+
+      flatTable().find("thead").should("exist");
+      flatTableBody().should("exist");
+    });
+
+    // Nested children causing linting error causes prevention of committing
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/flat-table/flat-table.test.js:2778]: Logged ticket FE-5613 to investigate the issue that prevents this case from running. Skipped test til it's resolved.

File: src/components/flat-table/flat-table.test.js [hunk 2]
```diff
@@ -0,0 +1,4635 @@
...
+      flatTableRowHeader()
+        .eq(0)
+        .find("span")
+        .should("have.attr", "data-component", "icon");
+    });
+
+    it("should render Flat Table Row Header with string as children", () => {
+      CypressMountWithProviders(<FlatTableSpanComponent />);
+
+      const rowHeaderText = "Code - Sticky ";
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/flat-table/flat-table.test.js:2869]: Probably not. Just a habit of declaring these as consts to keep the assertions neat.

File: src/components/flat-table/flat-table.test.js [hunk 3]
```diff
@@ -0,0 +1,4635 @@
...
+
+      flatTableRowHeader().eq(0).should("have.text", rowHeaderText);
+    });
+
+    it("should render Flat Table Cell with string as children", () => {
+      CypressMountWithProviders(<FlatTableTruncateBgComponent />);
+
+      const cellText1 = "John Doe";
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/flat-table/flat-table.test.js:2877]: As above.

File: src/components/flat-table/flat-table.test.js [hunk 4]
```diff
@@ -0,0 +1,4635 @@
...
+    });
+
+    it("should render Flat Table with sticky header", () => {
+      CypressMountWithProviders(
+        <div style={{ height: "150px" }}>
+          <FlatTableComponent hasStickyHead />
+        </div>
+      );
+
+      cy.wait(300, { log: DEBUG_FLAG }); // required because element needs to be loaded
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/flat-table/flat-table.test.js:2906]: I don't think so. I don't even remember putting that in!

**Review by DlgSHi — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**